packages/relay/src/lib/services/cacheService/cacheService.ts

```diff
@@ -217,35 +225,45 @@ export class CacheService {
 
   /**
    * Sets multiple values in the cache, each associated with its respective key.
+   * Depends on env. variable, whether we will be using pipelining method or multiSet
    * If the shared cache is enabled and an error occurs while setting in it,
    * the internal cache is used as a fallback.
    * @param {Record<string, any>} entries - An object containing key-value pairs to cache.
    * @param {string} callingMethod - The name of the method calling the cache.
+   * @param {number} ttl - Time to live for the cached value in milliseconds (optional).
    * @param {string} requestIdPrefix - A prefix to include in log messages (optional).
    * @param {boolean} shared - Whether to use the shared cache (optional, default: false).
    */
   public multiSet(
     entries: Record<string, any>,
     callingMethod: string,
+    ttl?: number,
     requestIdPrefix?: string,
     shared: boolean = true,
   ): void {
     if (shared && this.isSharedCacheEnabled) {
+      const metricsMethod = this.shouldMultiSet ? CacheService.methods.MSET : CacheService.methods.PIPELINE;
       try {
-        this.sharedCache.multiSet(entries, callingMethod, requestIdPrefix);
-        this.cacheMethodsCounter.labels(callingMethod, CacheService.cacheTypes.REDIS, CacheService.methods.MSET).inc(1);
+        if (this.shouldMultiSet) {
+          this.sharedCache.multiSet(entries, callingMethod, requestIdPrefix);
+        } else {
+          this.sharedCache.pipelineSet(entries, callingMethod, ttl, requestIdPrefix);
+        }
+
+        this.cacheMethodsCounter.labels(callingMethod, CacheService.cacheTypes.REDIS, metricsMethod).inc(1);
       } catch (error) {
         const redisError = new RedisCacheError(error);
         this.logger.error(
           `${requestIdPrefix} Error occurred while setting the cache to Redis. Fallback to internal cache. Error is: ${redisError.fullError}`,
         );
-        // Fallback to internal cache
-        this.internalCache.multiSet(entries, callingMethod, requestIdPrefix);
-        this.cacheMethodsCounter.labels(callingMethod, CacheService.cacheTypes.LRU, CacheService.methods.MSET).inc(1);
+        // Fallback to internal cache, but use pipeline, because of it's TTL support
+        this.internalCache.pipelineSet(entries, callingMethod, ttl, requestIdPrefix);
+        this.cacheMethodsCounter.labels(callingMethod, CacheService.cacheTypes.LRU, CacheService.methods.SET).inc(1);
```

Comment:
It's still a multi set thought right. Why the change to the metric label here and below?
If you change it you'll give the impression the SET was called instead

Reply:
Actually, if it fallbacks to internal cache, it will use `set`, because internal cache does not support `multi set`.